### PR TITLE
Add gpt docs support

### PR DESCRIPTION
**Pull request by beerose**
🚨 **IMPORTANT: Please do not create a Pull Request without creating an issue first.**

_Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request._

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Related # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots/Sandbox (if appropriate/relevant):

Adding links to sandbox or providing screenshots can help us understand more about this PR and take action on it as appropriate

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Environment**:

- OS:
- `@graphql-codegen/...`:
- NodeJS:

## Checklist:

- [ ] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: website/package.json [hunk 1]
```diff
@@ -52,20 +52,21 @@
...
     "@graphql-codegen/typescript-stencil-apollo": "2.3.6",
     "@graphql-codegen/typescript-type-graphql": "2.3.6",
-    "@graphql-codegen/typescript-nhost": "0.0.1",
     "@graphql-codegen/typescript-urql": "3.7.3",
     "@graphql-codegen/typescript-vue-apollo": "3.3.7",
     "@graphql-codegen/typescript-vue-apollo-smart-ops": "2.3.6",
     "@graphql-codegen/typescript-vue-urql": "2.3.6",
     "@graphql-codegen/urql-introspection": "2.2.1",
     "@graphql-codegen/client-preset": "4.0.0",
+    "@mendable/search": "^0.0.102",
```
Comment (beerose) [inline website/package.json:69]:
```suggestion
    "@mendable/search": "0.0.102",
```

File: .github/workflows/website.yml [hunk 2]
```diff
@@ -29,6 +29,7 @@ jobs:
         env:
           NEXT_BASE_PATH: ${{ github.ref == 'refs/heads/master' && '/graphql/codegen' || '' }}
           SITE_URL: ${{ github.ref == 'refs/heads/master' && 'https://the-guild.dev/graphql/codegen' || '' }}
+          NEXT_PUBLIC_MENDABLE_ANON_KEY: ${{ secrets.NEXT_PUBLIC_MENDABLE_ANON_KEY }}
```
Comment (beerose) [inline .github/workflows/website.yml:32]: Need someone to add it to github's secrets and Vercel

**Review by YassinEldeeb — CHANGES_REQUESTED** [review]
looks good!

I just found this issue, where the search bar wouldn't open up the dialog unless you clicked on the edges, but not in the center.


https://github.com/dotansimha/graphql-code-generator/assets/72823042/e82d5974-aed1-4071-bf58-93b0ea7ac9d6

**Comment by YassinEldeeb** [comment]
maybe, it's worth highlighting in the input placeholder that is is AI search, because people might bounce when asking questions and finding that it takes longer than the normal search like Algolia

**Comment by YassinEldeeb** [comment]
maybe it's worth changing either of the shortcuts of the two searches because I think currently CMD + K pops up both searches, normal and AI

image

**Review by saihaj — APPROVED** [review]
I suppose to disable `cmd+k` on the global search bar, that is a change we need to have configurable in our global docs theme

**Comment by beerose** [comment]
There's a prop in Mendable component that lets you override that. Will update it

**Review by dotansimha — APPROVED** [review]

**Comment by B2o5T** [comment]
@beerose can we hide ctrl+k in mentabable input? currenly it's open algolia search

File: website/theme.config.tsx [hunk 3]
```diff
@@ -2,9 +2,33 @@
...
 import { defineConfig, Giscus, useTheme } from '@theguild/components';
+import { MendableSearchBar } from '@mendable/search';
 
 export default defineConfig({
   docsRepositoryBase: 'https://github.com/dotansimha/graphql-code-generator/tree/master/website',
+  search: {
+    component: () => {
+      return (
+        <div className="hidden w-[250px] sm:block">
+          <MendableSearchBar
```
Comment (B2o5T) [inline website/theme.config.tsx:13]: seems we can remove the outer div? and add className directly to MendableSearchBar
image

Comment (YassinEldeeb) [inline website/theme.config.tsx:13]: the center or most of the search bar doesn't open the dialog on click, and it only happen on Firefox, it's fine with Chrome

image

**Comment by beerose** [comment]
> @beerose can we hide ctrl+k in mentabable input? currenly it's open algolia search
> 
> also can we adjust bg color of input on dark theme? to be dark and not light

AFAIK, cmd+K opens algolia

File: website/theme.config.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (beerose) [reply website/theme.config.tsx:13]: FYI that means loading a separate CSS file

**Comment by beerose** [comment]
> also can we adjust bg color of input on dark theme? to be dark and not light

No, not possible in the current version of Mendable

File: website/src/components/Search.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+// const darkModeStore = {
+//   subscribe(callback: () => void) {
+//     const mutationObserver = new MutationObserver(callback);
+//     mutationObserver.observe(document.documentElement, { attributeFilter: ['class'] });
+
+//     return () => mutationObserver.disconnect();
+//   },
+//   getSnapshot: () => document.documentElement.classList.contains('dark'),
+//   getServerSnapshot: () => false,
+// };
```
Comment (B2o5T) [inline website/src/components/Search.tsx:14]:
```suggestion
```

**Review by B2o5T — APPROVED** [review]

**Pull request merged**